### PR TITLE
Fix xarray groupby

### PR DESCRIPTION
**Pull request by emmaai**
Changes

- remove `xarray` version constraint
- fix the tests
- patch the docker image

I don't have a sensible way to test the docker image against the current code since it's supposed to be a regression test and nothing should be broken. Maybe I shouldn't have updated the docker image in this pr. I'll leave the docker image test failed until next pr specially on it.

### CODE REVIEW

**Comment by woodcockr** [comment]
@emmaai I can’t review this PR (outside my knowledge of the code) but I did want to say thanks for getting on top of this groupby issue and I’m looking forward to the update!

File: odc/stats/io.py [hunk 1]
```diff
@@ -229,8 +229,7 @@ def _apply_color_ramp(
 
         ows_style = StandaloneStyle(ows_style_dict)
         # assign the time to xr.Dataset cause ows needs it
```
Comment (Ariana-B) [inline odc/stats/io.py:231]: Seems like this comment is now outdated?

**Review by Ariana-B — APPROVED** [review]

File: odc/stats/io.py (hunk 1 above)
Comment (emmaai) [reply odc/stats/io.py:231]: the code still does what's in the comment, only differently, to be compatible with newer xarray

**Review by SpacemanPaul — APPROVED** [review]
All looks sensible to me.

I think the docker requirements might more work given the remaining failing GH action - although I can't see what is going wrong in the docker build.

**Pull request merged**